src/main/java/redis/clients/jedis/params/StrAlgoLCSParams.java

```diff
@@ -0,0 +1,98 @@
+package redis.clients.jedis.params;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+import redis.clients.jedis.Protocol;
+import redis.clients.jedis.util.SafeEncoder;
+
+public class StrAlgoLCSParams extends Params {
+
+    private static final String IDX = "idx";
+    private static final String LEN = "len";
+    private static final String WITHMATCHLEN = "withmatchlen";
+    private static final String MINMATCHLEN = "minmatchlen";
+    private static final String STRINGS = "strings";
+
+    public StrAlgoLCSParams() {
+    }
+
+    public static StrAlgoLCSParams StrAlgoLCSParams() {
+        return new StrAlgoLCSParams();
+    }
+
+    /**
+     * When IDX is given the command returns an array with the LCS length
+     * and all the ranges in both the strings, start and end offset for
+     * each string, where there are matches.
+     * @return StrAlgoParams
+     */
+    public StrAlgoLCSParams idx() {
+        addParam(IDX);
+        return this;
+    }
+
+    /**
+     * When LEN is given the command returns the length of the longest common substring.
+     * @return StrAlgoParams
+     */
+    public StrAlgoLCSParams len() {
+        addParam(LEN);
+        return this;
+    }
+
+    /**
+     * When WITHMATCHLEN is given each array representing a match will also have the length of the match.
+     * @return StrAlgoParams
+     */
+    public StrAlgoLCSParams withMatchLen() {
+        addParam(WITHMATCHLEN);
+        return this;
+    }
+
+    /**
+     * Specify the minimum match length.
+     * @return StrAlgoParams
+     */
+    public StrAlgoLCSParams minMatchLen(long minMatchLen) {
+        addParam(MINMATCHLEN, minMatchLen);
+        return this;
+    }
+
+    /**
+     * Specify the strings to be passed.
+     * @return StrAlgoParams
+     */
+    public StrAlgoLCSParams strings(String... strings) {
```

Comment:
Instead of this, strAlgoLcs should have two more (string) params.

Reply:
done

Reply:
@yangbodong22011 That's not what I meant. It's my mistake, sorry.

`strAlgoLcs` "method" should have two more string params.

Reply:
@sazzad16 you mean this?
```
public LCSMatchResult strAlgoLCS(StrAlgoLCSParams params, String stringA, String stringB);
```

Reply:
@yangbodong22011 yes

Reply:
@sazzad16 we can pass `String` or `byte[]` args through the following methods:
```
public StrAlgoLCSParams strings(String stringA, String stringB);
public StrAlgoLCSParams strings(byte[] stringA, byte[] stringB); (need to be added)
```

why we need this:
```
public LCSMatchResult strAlgoLCS(StrAlgoLCSParams params, String stringA, String stringB);
```

and when user call `strAlgoLCS(params.strings(a,b), c, d)`, will pass four strings to redis.

Reply:
@yangbodong22011 I meant removing `strings` option from Params class, just like removing keys option that you did earlier.

Reply:
@sazzad16 ohh, i got it and agree, let me update it.

Reply:
@sazzad16 done, now we have two commands: `strAlgoLCSKeys` and `strAlgoLCSStrings`.